src/applications/Job.java

```diff
@@ -0,0 +1,53 @@
+package applications;
+
+import dataStructures.LinkedQueue;
+
+class Job {
+    // data members
+    private LinkedQueue taskQ; // this job's tasks
+    private int length; // sum of scheduled task times
+    private int arrivalTime; // arrival time at current queue
+    private int id; // job identifier
+
+    // constructor
+    Job(int theId) {
+        id = theId;
+        taskQ = new LinkedQueue();
+        // length and arrivalTime have default value 0
+    }
+
+    // other methods
+    public void addTask(int theMachine, int theTime) {
+        getTaskQ().put(new Task(theMachine, theTime));
+    }
+
+    /**
+     * remove next task of job and return its time also update length
+     */
+    public int removeNextTask() {
+        int theTime = ((Task) getTaskQ().remove()).getTime();
+        length = getLength() + theTime;
+        return theTime;
+    }
+
+    public LinkedQueue getTaskQ() {
+        return taskQ;
+    }
+
+    public int getLength() {
+        return length;
+    }
+
+    public int getArrivalTime() {
+        return arrivalTime;
+    }
+
+    public void setArrivalTime(int arrivalTime) {
```

Comment:
We should see if we can eliminate this setter, or at least make it so that it's purpose (why we would want to overwrite the `arrivalTime`) is clearer.